### DISCUSSION TITLE
Static CSS import fails with ERR_FILE_NOT_FOUND in production build

### DISCUSSION THREAD

**Discussion by kokanek** [vercel/next.js#13578] [Help]
# Bug report
Everything works just fine when I run 
'next dev' or 'next start' 
Also, 'next export' creates the proper static files, where I can also see the correctly linked css bundle. But when I open index.html, I get error while importing static files.
<img width="613" alt="Screen Shot 2020-05-30 at 3 30 34 PM" src="https://user-images.githubusercontent.com/16059151/83325539-965bb300-a28a-11ea-8ee0-ea30296f4787.png">

Also, deployed the same to github pages using the deploy script: https://kokanek.github.io/portfolio/
```
"deploy": "next build && next export && git add out/ && git commit -m \"Deploy gh-pages\" && git subtree push --prefix out origin gh-pages"
```
But getting the same error there as well. 
<img width="765" alt="Screen Shot 2020-05-30 at 3 40 52 PM" src="https://user-images.githubusercontent.com/16059151/83325709-f868e800-a28b-11ea-93a2-375be619fc63.png">

## Describe the bug
I created a new project using create-next-app. 
Added the <code>_app.js</code> inside the pages folder. 
Also, added a <code>styles.css</code> file and imported it in the _app.js
###  _app.js
```
import '../styles.css'

export default function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}
```

The styles.css file contains some basic css:
```
html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
    Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
    sans-serif;
}

* {
  box-sizing: border-box;
}
.container {
  min-height: 100vh;
  padding: 0 0.5rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

main {
  padding: 5rem 0;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

footer {
  width: 100%;
  height: 100px;
  border-top: 1px solid #eaeaea;
  display: flex;
  justify-content: center;
  align-items: center;
}

footer img {
  margin-left: 0.5rem;
}

footer a {
  display: flex;
  justify-content: center;
  align-items: center;
}
...
<some-more-css>
...
```
Here is the config.js
Added the next-css dependency in order to make it work but it doesn't seem to help.
### next.config.js
```
const withCSS = require('@zeit/next-css')
module.exports = withCSS({
  cssModules: true
})
```

## To Reproduce

1. Clone this repository: https://github.com/kokanek/portfolio
2. checkout the master branch
3. Run the 'npm run build' and 'npm run export' script.
4. Check the /out folder and open the index.html to see the errors.

## Expected behavior
The static files generated in the /out folder should work the same way as the build deployment

## System information

- OS: [e.g. macOS, Windows]
Macos
- Browser (if applies) [e.g. chrome, safari]
Chrome
- Version of Next.js: [e.g. 6.0.2]
9.4.2
- Version of Node.js: [e.g. 10.10.0]

**Comment by jamesmosier** [comment]
Are you just trying to open the `out/index.html` file in your browser on your local computer? If so, this won't work. You need to serve these files on some sort of HTTP server in order for it to load those scripts.

As for why it doesn't work on GitHub pages, your site's url is https://kokanek.github.io/portfolio but Next is looking for scripts at `https://kokanek.github.io/_next/static/css/d4fed0ecf2bef9b3dbc2.css` (notice no `/portfolio` path).

**Reply by kokanek** [reply]
Hey @jamesmosier Thanks for the reply. To fix the issue that you pointed out, 
I have added an assetPrefix path in next.config.js:
```
module.exports = {
  assetPrefix: !debug ? '/portfolio/' : '',
}
```
and also, used this deployment script which creates the .nojekyll file on every deployment:
```
"deploy": "rm -rf node_modules/.cache && next build && next export && touch out/.nojekyll && git add out/ && git commit -m \"Deploy Next.js to gh-pages\" && git subtree push --prefix out origin gh-pages"
```
& that seems to have sorted the issue. 

### But, it created a new one:
My build on vercel started to break after this, though, because it started to look for /portfolio even though I had deployed on kokaneka.now.sh. So I had to pass a node environment as:
NODE_ENV = 'nextjs' so that the assetPrefix of "" (and not 'portfolio') gets assigned in vercel deployment. I am not sure if this was the right way to fix it. 

Let me know if you have any clue.

**Reply by jamesmosier** [reply]
You should not overwrite the `NODE_ENV`, but you could set a different environment variable (i.e. `PORTFOLIO_ASSET_PREFIX`) and do the same thing as you are doing now.

**Comment by puntope** [comment]
`module.exports = {
	assetPrefix: '.',
}`

**Reply by tsevdos** [reply]
You are a life saver!!!! Thank you!!!!

**Reply by Linedev-Web** [reply]
Thank you, I've had a MIME TYPE (text/html) problem for 2 days now.
assertPrefix has just corrected the MIME TYPE problem for me, plus the local redirections when I give the project to my friend.
I looked in the doc https://nextjs.org/docs/app/api-reference/next-config-js/assetPrefix except I didn't tell myself to just put '.' to solve my problem.

Thanks

**Comment by choochiutrainz** [comment]
Hey folks, have y'all found a solution? I've run into the same issue. My website can deploy, but there's no CSS in it, because Next can't find https://choochiutrain.com/_next/static/css/66453a5667fe41355a7e.css. But looking at the repo, it's quite clearly where it's supposed to be :/

Any help much appreciated. Thank you

**Reply by choochiutrainz** [reply]
just kidding! We're good, disregard. Thank you
